core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java

```diff
@@ -631,6 +631,19 @@ private static void generateConfigBuilder(
             }
 
             for (ConfigClassWithPrefix mapping : mappings) {
+                boolean isPresentAtDeployment = isClassAvailable(deploymentCl,
```

Comment:
This should be checked in https://github.com/quarkusio/quarkus/blob/be4817d541882225767e9e3655ec5bf325045360/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java#L98-L111.

Reply:
Or better yet in here:
https://github.com/quarkusio/quarkus/blob/be4817d541882225767e9e3655ec5bf325045360/core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java#L166-L184

Which already reports other checks.